### PR TITLE
Fix Baton Pass Clause for Enchanted Items

### PR DESCRIPTION
**Pull request by Spandan7**
Okay now Dusk Ball is recognized as an item which boosts speed. 
Thanks again to Quite Quiet

### CODE REVIEW

**Comment by Spandan7** [comment]
This one @TheImmortal

File: mods/enchanteditems/rulesets.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+			// check if Speed is boosted
+			let speedBoosted = false;
+			for (let i = 0; i < set.moves.length; i++) {
+				let move = this.getMove(set.moves[i]);
+				if (move.boosts && move.boosts.spe > 0) {
+					speedBoosted = true;
+					break;
+				}
+			}
+			let boostSpeed = ['duskball', 'flamecharge', 'geomancy', 'motordrive', 'rattled', 'speedboost', 'steadfast', 'weakarmor', 'salacberry'];
```
Comment (urkerab) [inline mods/enchanteditems/rulesets.js]: `salacberry` is Justified, so doesn't boost speed, and you haven't added the items for Motor Drive, Rattled, Steadfast or Weak Armor.

File: mods/enchanteditems/rulesets.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+			// check if non-Speed boosted
+			let nonSpeedBoosted = false;
+			for (let i = 0; i < set.moves.length; i++) {
+				let move = this.getMove(set.moves[i]);
+				if (move.boosts && (move.boosts.atk > 0 || move.boosts.def > 0 || move.boosts.spa > 0 || move.boosts.spd > 0)) {
+					nonSpeedBoosted = true;
+					break;
+				}
+			}
+			let boostNonSpeed = ['acupressure', 'starfberry', 'curse', 'metalclaw', 'meteormash', 'poweruppunch', 'rage', 'rototiller', 'fellstinger', 'bellydrum', 'download', 'justified', 'moxie', 'sapsipper', 'defiant', 'angerpoint', 'cellbattery', 'liechiberry', 'snowball', 'weaknesspolicy', 'diamondstorm', 'flowershield', 'skullbash', 'steelwing', 'stockpile', 'cottonguard', 'ganlonberry', 'keeberry', 'chargebeam', 'fierydance', 'geomancy', 'lightningrod', 'stormdrain', 'competitive', 'absorbbulb', 'petayaberry', 'charge', 'apicotberry', 'luminousmoss', 'marangaberry'];
```
Comment (urkerab) [inline mods/enchanteditems/rulesets.js]: ... and these are wrong too. Full list at http://www.smogon.com/forums/posts/6877561

**Comment by Spandan7** [comment]
Wew I keep forgetting things thanks

**Pull request merged**